### PR TITLE
fix: Reapply Update if CronWorkflow resource changed

### PR DESCRIPTION
**Pull request by simster7**

### CODE REVIEW

File: workflow/cron/operator.go [hunk 1]
```diff
@@ -115,8 +119,54 @@ func getWorkflowObjectReference(wf *v1alpha1.Workflow, runWf *v1alpha1.Workflow)
...
+			return err
+		}
+		var newCronWf v1alpha1.CronWorkflow
+		err = json.Unmarshal(newCronWfBytes, &newCronWf)
+		if err != nil {
+			return err
+		}
+		_, err = woc.cronWfIf.Update(&newCronWf)
+		if err == nil {
+			return nil
```
Comment (alexec) [inline workflow/cron/operator.go:166]: do we want to return the error on attempt 5?

Comment (simster7) [reply workflow/cron/operator.go:166]: We do. After attempt 5 the loop breaks and returns an error on line 169

File: workflow/cron/operator.go [hunk 2]
```diff
@@ -115,8 +119,54 @@ func getWorkflowObjectReference(wf *v1alpha1.Workflow, runWf *v1alpha1.Workflow)
...
+	}
+	curr, err := json.Marshal(woc.cronWf)
+	if err != nil {
+		return err
+	}
+	patch, err := jsonpatch.CreateMergePatch(orig, curr)
+	if err != nil {
+		return err
+	}
+	for i := 0; i < 5; i++ {
```
Comment (sarabala1979) [inline workflow/cron/operator.go:146]: Instead of loop, you can use `wait.ExponentialBackoff`

Comment (simster7) [reply workflow/cron/operator.go:146]: Good suggestion. I actually considered this but decided to go with this approach for consistency with the Workflow operator

Comment (simster7) [reply workflow/cron/operator.go:146]: If needed, we can change both in a separate PR

File: workflow/cron/operator.go (hunk 1 above)
Comment (alexec) [reply workflow/cron/operator.go:166]: sorry - not "an error" - "the error" i.e. the error returned by `Update`

Comment (simster7) [reply workflow/cron/operator.go:166]: Sure, I see what you mean. Will refactor

**Pull request merged**